Engine/source/gfx/D3D11/gfxD3D11TextureArray.cpp

```diff
@@ -0,0 +1,266 @@
+            tex->GetDesc(&desc);
+            if (desc.Height != texElementDesc.Height || desc.Width != texElementDesc.Width || textureArray[idx].getFormat() != format)
+            {
+               if (desc.Height != texElementDesc.Height || desc.Width != texElementDesc.Width)
+               {
+                  AssertWarn(true, "GFXD3D11TextureArray::fromTextureArray all textures should be the same size");
+                  Con::warnf("GFXD3D11TextureArray::fromTextureArray all textures should be the same size");
+               }
+               else
+               {
+                  AssertWarn(true, "GFXD3D11TextureArray::fromTextureArray all textures should have the same format");
+                  Con::warnf("GFXD3D11TextureArray::fromTextureArray all textures should have the same format");
+               }
+
+               GBitmap* inBitmap = TEXMGR->loadUncompressedTexture(textureArray[idx]->getPath(), &GFXTexturePersistentProfile, texElementDesc.Width, texElementDesc.Height);
+               if (!inBitmap->setFormat(format))
+               {
+                  AssertWarn(true, "GFXD3D11TextureArray::fromTextureArray all textures must be convertible to GFXFormatR8G8B8A8");
+                  Con::errorf("GFXD3D11TextureArray::fromTextureArray all textures must be convertible to GFXFormatR8G8B8A8");
+                  success = false;
+                  handle = NULL;
+                  delete inBitmap;
+               }
+               else
+               {
+                  handle = TEXMGR->createTexture(inBitmap, "", &GFXStaticTextureProfile, true);
+                  tmpHandles.push_back(handle);
+               }
+            }
+         }
+         if (handle.isValid())
+         {
+            D3D11_TEXTURE2D_DESC desc;
+            ID3D11Texture2D* tex = dynamic_cast<GFXD3D11TextureObject*>(handle.getPointer())->get2DTex();
+            tex->GetDesc(&desc);
+            if (desc.Height != texElementDesc.Height || desc.Width != texElementDesc.Width || handle.getFormat() != format)
+            {
+               AssertWarn(true, "GFXD3D11TextureArray::fromTextureArray all textures must have the same size and format");
+               Con::errorf("GFXD3D11TextureArray::fromTextureArray all textures must have the same size and format");
+               success = false;
+            }
+            texture2Ds[idx] = dynamic_cast<GFXD3D11TextureObject*>(handle.getPointer())->get2DTex();
+         }
+      }
+   }
+
+   D3D11_TEXTURE2D_DESC texArrayDesc;
+   texArrayDesc.Width = texElementDesc.Width;
+   texArrayDesc.Height = texElementDesc.Height;
+   texArrayDesc.MipLevels = texElementDesc.MipLevels;
+   texArrayDesc.ArraySize = mArraySize;
+   texArrayDesc.Format = GFXD3D11TextureFormat[format];
+   texArrayDesc.SampleDesc.Count = 1;
+   texArrayDesc.SampleDesc.Quality = 0;
+   texArrayDesc.Usage = D3D11_USAGE_DEFAULT;
```

Comment:
D3D11_USAGE_DEFAULT can't be used with the cpu access flag D3D11_CPU_ACCESS_WRITE. See here https://docs.microsoft.com/en-us/windows/win32/api/d3d11/ne-d3d11-d3d11_cpu_access_flag

Reply:
Turns out I could remove the CPU_ACCESS flag completely, nice catch!